src/dune_rules/install_rules.ml

```diff
@@ -499,6 +499,21 @@ end = struct
     Memo.exec memo
 end
 
+module All_sctx = struct
+  let all_sctx_to_entries (scontexts : Super_context.t Context_name.Map.t) :
+      Install.Entry.Sourced.t list Package.Name.Map.t Context_name.Map.t Memo.t
+      =
+    let open Memo.O in
+    let ctxs = Context_name.Map.to_list scontexts in
+    let* pairs =
+      Memo.sequential_map ctxs ~f:(fun (ctxn, sctx) ->
+          let* packages = Stanzas_to_entries.stanzas_to_entries sctx in
+          Memo.return (ctxn, packages))
+    in
+    let maps = Context_name.Map.of_list_reduce pairs ~f:(fun old _new -> old) in
```

Comment:
The reduce doesn't seem right. How could you have a collision on a context name here? Shouldn't it be `Map.of_list_exn`?

Reply:
I believe that of_list_reduce works correctly, but of_list_exn is more naturally the right function to use, so I will change to use it.